### PR TITLE
Add examples of proposed new APIs

### PR DESCRIPTION
**Pull request by avtakkar**

### CODE REVIEW

File: docs/distribution/README.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+    #### Sample Request
+    ```
+    PUT https://localhost:6000/v2/hello-world/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/signatures/sha256:2235d2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49cc77c
+    ```
+
+### Option 2: Server-Side Signature Linking
+
+The server detects a signature push and implicitly links the signature to the target.
+#### TODO
```
Comment (avtakkar) [inline docs/distribution/README.md:98]: @SteveLasker , we haven't spoken about what the API will look like when the server handles linking of signatures implicitly

File: docs/distribution/README.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+
+**Cons with this approach:**
+
+* OCI Index would need to support a config object:  [Add Index Support for Artifact Type #806](https://github.com/opencontainers/image-spec/issues/)
+
+## Signature Discovery
+
+### Signature Discovery - Option 1: distribution-spec consistent paging
+
+The [OCI distribution-spec][distribution-spec-paging] identifies paging with `n` and `last` parameters:
```
Comment (avtakkar) [inline docs/distribution/README.md:219]: This is option 2 below

**Comment by SteveLasker** [comment]
I think we can close this one in lieu of: https://github.com/notaryproject/nv2/pull/5

**Comment by avtakkar** [comment]
See https://github.com/notaryproject/nv2/pull/5

**Pull request closed without merging**